testproject/Assets/Tests/Runtime/MultiprocessRuntime/ExecuteStepInContextTests.cs

```diff
@@ -142,31 +141,70 @@ public IEnumerator ContextTestWithAdditionalWait()
             InitializeContextSteps();
 
             const int maxValue = 10;
+            const float resendResultTime = 2.0f;
+            bool exitOnFailure = false;
+            int count = 0;
+            bool sendNextResult = true;
+            float lastResultSentTime = 0;
             yield return new ExecuteStepInContext(StepExecutionContext.Clients, _ =>
             {
-                int count = 0;
-
                 void UpdateFunc(float _)
                 {
-                    TestCoordinator.Instance.WriteTestResultsServerRpc(count++);
-                    if (count > maxValue)
+                    if (!exitOnFailure)
                     {
-                        NetworkManager.Singleton.gameObject.GetComponent<CallbackComponent>().OnUpdate -= UpdateFunc;
+                        // We will send if we have another result to send =or= if we haven't received a response back from the server yet.
+                        if (sendNextResult || lastResultSentTime < Time.realtimeSinceStartup)
+                        {
+                            if (!sendNextResult && lastResultSentTime < Time.realtimeSinceStartup)
+                            {
+                                Debug.LogWarning($"{nameof(ContextTestWithAdditionalWait)} is resending due to timeout receiving result received response for count value = {count}!");
+                            }
+
+                            TestCoordinator.Instance.WriteTestResultsServerRpc(count);
+                            lastResultSentTime = resendResultTime + Time.realtimeSinceStartup;
+                            sendNextResult = false;
+                        }
```

Comment:
I'm a bit confused by these lines.
If there's an RPC that fails sending, that should fail the test, there shouldn't be more retries, especially with a reliable RPC.
If that's needed, it'd be more useful to add these checks to ExecuteStepInContext, or even to TestCoordinator, so that all future multiprocess tests benefit from this boilerplate retry?

Reply:
This code could come out but was just there in the event that the server was missing messages.

Reply:
That's my point above. If the server is missing messages, with reliable RPCs, that should fail the test.

Reply:
If the server is missing messages, it means something is terribly wrong.
These basic tests are like Canaries. If they fail, it's very likely a lot of other tests will fail and that something is wrong at the core of Netcode.

Reply:
Exactly... but there was no type of check for that until now?
If you fail a test but have no reason as to why you failed the test...then how do you know why you failed the test?

Reply:
there was a check for that. With the nbFinished check below, you're are sure there was RPC exchange back and forth (since RPCs are used to trigger these actions).
Adding this check here just makes this test fail silently.

Reply:
bump

Reply:
We shouldn't have tests that should fail, pass because we're giving them too many chances.

Reply:
@NoelStephensUnity and @SamuelBellomo what should we do with this change

Reply:
I'm inclined to agree with Sam here. Even if we can't immediately debug what was the issue causing tests to fail, I think we should still fail here instead of having retry attempts. Making tests fragile is generally a good idea instead of making them artificially robust by defensive checks, retrys and suppressions.
I'd vote to kill retry code even though I reckon it might make things harders to pass, which is a good thing IMO.

Reply:
Just to get everyone on the same page, we went over these in private chat last week with Noel and agreed to remove these.
I think Zain is waiting for the SDK to stabilize a bit more (and for his return from vacation :P ) to continue work on this PR

Reply:
@zain-mecklai I thought we agreed in our chat these would be removed? @JesseOlmer is saying you're ready to merge?

Reply:
@zain-mecklai
If you want me to assist with this I can as I was the one who added this code originally.